aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,109 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef axis, int64_t spacing_size,  int64_t edge_order) {
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 as input.");
+  TORCH_CHECK( spacing_size == axis.size(), "torch.gradient should receive axis and spacing size to be equal.");
+  TORCH_CHECK( edge_order == 1, "torch.gradient supports edge_order value only to be equal to 1.");
+  dim_list_to_bitset(axis, self.dim());
+}
+
+std::vector<Tensor> gradient_helper(const Tensor& self, TensorList spacing, IntArrayRef axis, int64_t edge_order) {
+  for (const auto i : c10::irange(spacing.size())) {
+    TORCH_CHECK(self.device() == spacing[i].device(), "torch.gradient requires each tensor in spacing to be in the same device as self.");
+  }
+  pre_check_gradient(self, axis, spacing.size(), edge_order);
+
+  std::vector<Tensor> result;
+  for (const auto i : c10::irange(axis.size())) {
+    TORCH_CHECK( spacing[i].dim() < 2, "torch.gradient's argument spacing can receive eiter a scalar, a list of scalars or a list of one-dimensional tensors");
```

Comment:
wouldn't this be part of the pre-check, too? Or are these checks here because they only apply when spacing is a tensorlist, and this is trying to make it easy to reuse pre-check between the list of scalars and tensor list versions of the operator?

Reply:
Yes, actually these checks are only for tensorlist and that's why it's not part of pre-check :)